### PR TITLE
Load enpoints data in frontend

### PR DESCRIPTION
**Pull request by KiviGibson**
### Goal
Create a custom hook and use axios to get the data from the api endpoint

### Trello
https://trello.com/c/n3fI6iO0/

### CODE REVIEW

File: frontend/App.tsx [hunk 1]
```diff
@@ -1,10 +1,15 @@
 import { StatusBar } from 'expo-status-bar';
 import { StyleSheet, Text, View } from 'react-native';
-
+import useFetch from './utils/useFetch';
+import data from "./api/endpoint.json"
 export default function App() {
+  const [val, callUrl] = useFetch({url:data.baseApiUrl+data.getRegistrationForm.url,method:data.getRegistrationForm.method[0]});
```
Comment (bartek3xe) [inline frontend/App.tsx:6]: We need to specify that getRegistrationForm.method should be an array or not. If not, we should add a key for all available methods. For example

`  "getRegistrationForm": {
    "url": "/registration/form/get",
    "method": {
    "GET": "GET",
  }`

File: frontend/App.tsx [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
+  function newCall(){
+    callUrl({})
+  }
   return (
     <View style={styles.container}>
-      <Text>Open up App.js to start working on your app!</Text>
+      <Text>{val}</Text>
       <StatusBar style="auto" />
     </View>
   );
```
Comment (bartek3xe) [inline frontend/App.tsx:15]: It's good to test our code, but remember: after that we should remove useFetch from this file.

File: frontend/utils/useFetch.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { useEffect, useState } from "react";
+import axios from "axios";
+const useFetch = (options:{}) =>{
```
Comment (bartek3xe) [inline frontend/utils/useFetch.tsx:3]: Define a custom type for your JSON data before

File: frontend/utils/useFetch.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    const [json, setJson] = useState("");
+
+    function call(option:{}){
+          axios.request(option).then(function (response:any){
+            setJson(response.data);
```
Comment (bartek3xe) [inline frontend/utils/useFetch.tsx:8]: Make sure to type the response data using custom type

File: frontend/utils/useFetch.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+          }).catch(function (error:string) {
+            console.error(error);
```
Comment (bartek3xe) [inline frontend/utils/useFetch.tsx:10]: We might consider more advanced error handling here

**Review by bartek3xe — CHANGES_REQUESTED** [review]

File: frontend/utils/useFetch.tsx (hunk 4 above)
Comment (KiviGibson) [reply frontend/utils/useFetch.tsx:8]: But this function returns just json file of any kind. We will choose later what type is it based on context

File: frontend/utils/useFetch.tsx (hunk 3 above)
Comment (KiviGibson) [reply frontend/utils/useFetch.tsx:3]: I think we can use types for validating data like interfaces, but i dont think all data will look the same so we can't define type of returned json data.
Egz.
registration form will differ from user profile data.

File: frontend/App.tsx (hunk 2 above)
Comment (KiviGibson) [reply frontend/App.tsx:15]: will do chief!

File: frontend/utils/useApi.tsx
Comment (KiviGibson) [inline frontend/utils/useApi.tsx:1]: its not finished,
added function to validate endpoint and its method for extra security, created object to have better way of manipulating this hook

File: frontend/utils/useFetch.tsx (hunk 4 above)
Comment (bartek3xe) [reply frontend/utils/useFetch.tsx]: My comment refers to the line with setJson(response.data);

File: frontend/utils/useApi.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+      setData(response.data)      
+    }).catch(function (error) {
+      throw new Error(error);
+    })
+  }
+
+  function checkForEndpoint(end:string,method:string): string
+  {
+    
+    if(!(end in endpoints)) throw new Error(`No endpoint found! Avileble: ${JSON.stringify(endpoints)}`);
```
Comment (bartek3xe) [inline frontend/utils/useApi.tsx:32]: A wrong endpoint should also return JSON, for example, `{"error": "content of the error"}`, but this is something to handle on the backend. Then, we can consider improving the error handling on the front.

**Review by bartek3xe — CHANGES_REQUESTED** [review]

**Review by bartek3xe — APPROVED** [review]

**Pull request merged**